cypress/integration/onlySubscribersCanRate.feature.js

```diff
@@ -0,0 +1,133 @@
+describe('Only subscribers can use rating functionality', () => {
+  beforeEach(() => {
+    cy.intercept('GET', 'https://fakest-newzz.herokuapp.com/api/articles', {
+      fixture: 'articles.json',
+    });
+    cy.intercept('GET', 'https://fakest-newzz.herokuapp.com/api/articles/4', {
+      fixture: 'specificArticle.json',
+    });
+    cy.intercept('POST', 'https://api.stripe.com/v1/tokens', { id: '1304124' });
+    cy.intercept(
+      'POST',
+      'https://fakest-newzz.herokuapp.com/api/subscriptions',
+      { paid: 'true', message: 'Thank you for subscribing' }
+    );
+    cy.intercept('POST', 'https://fakest-newzz.herokuapp.com/api/auth', {
+      fixture: 'registration.json',
+    });
+    cy.visit('/');
+  });
+
+  describe('Successfully', () => {
+    beforeEach(() => {
+      cy.intercept('POST', 'https://fakest-newzz.herokuapp.com/api/ratings', {
+        statusCode: 201,
+      });
+    });
+
+    it('is expected to restrict non-subscribers', () => {
+      cy.visit('/articles/4');
+      cy.get('[data-cy=article-container]').within(() => {
+        cy.get('[data-cy=article-rating]').should('contain', '3.7');
+        cy.get('[data-cy=article-rating-button]').find('i').eq(4).click();
+      });
+      cy.get('#rating-message').should(
+        'contain',
+        'You have to subscribe to be able to rate'
+      );
+    });
+
+    it('is expected to allow subscribers to rate', () => {
+      cy.get('[data-cy=login-button]').click();
+      cy.url('/registration');
```

Comment:
what does this do?

Reply:
Checks if the url contains /registration